### PR TITLE
[DEVELOPER-5677] Only require revision_log on published entities

### PR DESCRIPTION
**Pull request by rogeralloymagnetic**
### JIRA Issue Link
https://issues.jboss.org/browse/DEVELOPER-5677

### Verification Process
Create new node and confirm that Revision Log is **not** required
Edit Draft node and confirm that Revision Log is **not** required
Edit Published node and confirm that Revision Log **is** required
Edit Archived node and confirm that Revision Log is **not** required

#### Archived
screen shot 2019-02-20 at 11 44 39 am

#### Draft/New
screen shot 2019-02-20 at 11 43 39 am

#### Published
screen shot 2019-02-20 at 11 43 49 am

### CODE REVIEW

**Comment by robpblake** [comment]
retest this please

**Comment by jordanpagewhite** [comment]
retest this please

**Comment by robpblake** [comment]
@rogeralloymagnetic This is going to need master merged in to pick up changes to the tests.

We'll try and work out why this is suddenly happening.

cheers,

Rob

**Comment by rogeralloymagnetic** [comment]
@robpblake this is passing tests with the latest master merge. Not sure who's best to review this to get it ready to merge.

**Comment by jordanpagewhite** [comment]
I'll take a look

**Comment by jordanpagewhite** [comment]
I am happy with the code changes in this PR, and I tried the different scenarios in the PR environment and they worked as you reported. I have asked Joe to double-check the functionality and/or acceptance criteria on the ticket though.

**Comment by robpblake** [comment]
@rogeralloymagnetic Yeah for Drupal stuff I think @jordanpagewhite is generally your man. I know enough, but Jordan's Drupal knowledge is much better than mine.

cheers,

Rob

**Comment by jordanpagewhite** [comment]
@rogeralloymagnetic @LightGuard Yes, I agree with Jason's comment, which is what prompted me to request more thorough acceptance criteria from Joe. He updated the ticket yesterday with more thorough review steps. I am just double-checking one thing with him, and then I should be able to give this a proper review.

**Comment by jordanpagewhite** [comment]
I just finished up discussing this with Joe in Slack, and he agrees that the dynamically conditional behavior of that moderation state dropdown is more desirable. @rogeralloymagnetic Is this something you could achieve using Drupal Form API states?

**Comment by RubyGemTSL** [comment]
retest this please

**Comment by jordanpagewhite** [comment]
retest this please

File: _docker/drupal/drupal-filesystem/web/modules/custom/rhd_common/rhd_common.form_validations.inc [hunk 1]
```diff
@@ -49,13 +49,40 @@ function rhd_common_form_alter(&$form, &$form_state, $form_id) {
...
   }
 
-  // Require a revision log message.
-  if (isset($form['revision_log'])) {
+  // Require a revision log message for published nodes.
+  if (isset($form['revision_log']) && isset($form['moderation_state'])) {
+    $form['#validate'][] = 'check_moderation_state_validations';
+
+    $entity = $form_state->getFormObject()->getEntity();
+    $is_published = ($entity->get('moderation_state')->getValue()[0]['value'] === 'published');
```
Comment (jordanpagewhite) [inline _docker/drupal/drupal-filesystem/web/modules/custom/rhd_common/rhd_common.form_validations.inc]: Linking for posterity: https://issues.jboss.org/browse/DEVELOPER-5743

**Pull request merged**